MainWindow.cpp

```diff
@@ -139,7 +139,24 @@ MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), _ui(new Ui::MainW
 #ifdef RGM_SERVER_ENABLED
   RGMPlugin *pluginServer = new ServerPlugin(*this);
   auto outputTextBrowser = this->_ui->outputTextBrowser;
-  connect(pluginServer, &RGMPlugin::LogOutput, outputTextBrowser, &QTextBrowser::append);
+  connect(pluginServer, &RGMPlugin::LogOutput, [=](const QString& text, const QTextCharFormat &format) {
+    int startPos = 0;
+    int crPos = -1;
+    while ((crPos = text.indexOf('\r', startPos)) >= 0)  {
+        if (text.size() > crPos + 1 && text.at(crPos + 1) == '\n') {
+            outputTextBrowser->textCursor().insertText(text.mid(startPos, crPos - startPos) + '\n', format);
+            startPos = crPos + 2;
+            continue;
+        }
+        outputTextBrowser->textCursor().insertText(text.mid(startPos, crPos - startPos), format);
+        outputTextBrowser->textCursor().clearSelection();
+        outputTextBrowser->textCursor().movePosition(QTextCursor::StartOfBlock, QTextCursor::KeepAnchor);
+        startPos = crPos + 1;
+    }
+    if (startPos < text.count())
+        outputTextBrowser->textCursor().insertText(text.mid(startPos), format);
+  });
+  

```

Comment:
If you hooked this to output text browser's insert signal or w/e then ANSI escapes would be colored for all plugins instead of just the server plugin.